src/common/redis_module/ray_redis_module.cc

```diff
@@ -908,6 +912,90 @@ int TableTestAndUpdate_RedisCommand(RedisModuleCtx *ctx,
   return result;
 }
 
+/**
+ * Send notifications to the clients which subcribe this object. This function
+ * is divided into two parts: publishing notifications in broadcast channel and 
+ * publishing notifications in unicast channel. This function should be called 
+ * when add an object, delete an object, or the object manager list changed.
+ *
+ * @param ctx The Redis context.
+ * @param object_id The object ID of interest.
+ * @param table_key The opened key for the entry in the object table corresponding
+ *        to the object ID of interest.
+ * @param deleted The flag which indicates that the object has been removed. Maybe
+ *        some entries about this object have been deleted. In this case, some fields
+ *        of the notification protocol message will be filled in NULL or zero, instead of
+ *        getting some meta in db (E.g data size or object manager list).
+ * @return True if the publish was successful and false otherwise.
+ */
+int sendObjectNotification(RedisModuleCtx *ctx,
+                           RedisModuleString *object_id,
+                           RedisModuleKey *table_key, 
+                           bool deleted) {
+  RedisModuleString *data_size;
+  if (deleted) {
```

Comment:
This is the only place where we use the `deleted` argument. Are you saying that in the deleted case, sometimes the size information has been removed? I don't think we ever remove it.

I think the size information should still be around (and we should use the true `data_size` below instead of `0`).

That way we can remove the `deleted` argument.